src/test-utilities/react-testing.tsx

```diff
@@ -1,30 +1,52 @@
 import React from 'react';
 import {createMount} from '@shopify/react-testing';
-// eslint-disable-next-line shopify/strict-component-boundaries
-import {
-  createPolarisContext,
-  AppProviderContext,
-} from '../components/AppProvider';
+import {ClientApplication} from '@shopify/app-bridge';
+import {createPolarisContext} from '../utilities/create-polaris-context';
 // eslint-disable-next-line shopify/strict-component-boundaries
 import {FrameContext, FrameContextType} from '../components/Frame';
-// eslint-disable-next-line shopify/strict-component-boundaries
 import {
   createThemeContext,
   ThemeProviderContextType,
   ThemeProviderContext,
-} from '../components/ThemeProvider';
+} from '../utilities/theme';
+import {
+  ScrollLockManager,
+  ScrollLockManagerContext,
+} from '../utilities/scroll-lock-manager';
+import {StickyManager, StickyManagerContext} from '../utilities/sticky-manager';
+import {
+  createAppBridge,
+  AppBridgeContext,
+  AppBridgeOptions,
+} from '../utilities/app-bridge';
+import {I18n, I18nContext, TranslationDictionary} from '../utilities/i18n';
+import translations from '../../locales/en.json';
 import {PolarisContext} from '../components/types';
-import {DeepPartial} from '../types';
+import {DeepPartial, Discard} from '../types';
 import {merge} from '../utilities/merge';
+import {Link, LinkContext, LinkLikeComponent} from '../utilities/link';
 
-interface Providers {
+interface ComplexProviders {
   polaris: PolarisContext;
   themeProvider: ThemeProviderContextType;
   frame: FrameContextType;
 }
-type Options = DeepPartial<Providers>;
-type Context = Providers;
-interface Props extends Providers {
+interface SimpleProviders {
+  intl: TranslationDictionary | TranslationDictionary[];
+  scrollLockManager: ScrollLockManager;
+  stickyManager: StickyManager;
+  appBridge: AppBridgeOptions;
+  link: LinkLikeComponent;
+}
+type ReturnedContext = ComplexProviders &
```

Comment:
Can we rework this so we build up form multiple types instead of using Discard?

Reply:
I can, it'll make it a little more verbose though